internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -280,6 +287,36 @@ func (f *FleetGateway) convertToCheckinComponents(components []runtime.Component
 		checkinComponents = append(checkinComponents, checkinComponent)
 	}
 
+	// OTel status is placed as a component for each top-level component in OTel
+	// and each subcomponent is a unit.
+	if otelStatus != nil {
```

Comment:
I don't understand how this would be broken in Fleet. This is sending it exactly in the same format that Fleet understand today, so this just works transparently without any changes to Fleet. Which is what we discussed and what was implemented here.

Reply:
Fleet won't reject it, I am worried about how Fleet maps inputs back to integrations, since Fleet is integration focused and agent is only inputs.

We would need to test an integration implemented with Beats receivers and make sure the status reporting still works the same way.

Reply:
I didn't know there was a link between those two. I was just under the impression that the UI just rendered the content of the `components` field that is shipped to Fleet. If it does some referencing between the two (which I don't think it does), then it would be better to just let Fleet understand the `component` field.

Being this is experimental that is something that can easily change in the future.

Reply:
I think it looks into the meta information in the inputs, possibly it only needs the input ID to be what was in the agent policy for the component and it could figure it out.

I agree this isn't blocking and is something we could check easily with a test.

I don't love the `elastic-agent status output=json` format diverging from what gets sent to Fleet, I have used the fact that it is in the checkin format a few times for debugging, but I won't block the PR on it.